core/src/main/java/hudson/model/queue/WorkUnitContext.java

```diff
@@ -67,8 +67,8 @@ public WorkUnitContext(BuildableItem item) {
         this.item = item;
         this.task = item.task;
         this.future = (FutureImpl)item.getFuture();
-        this.actions = new ArrayList<Action>(item.getAllActions());
-        
+        // JENKINS-51584 do not use item.getAllActions() here.
+        this.actions = new ArrayList<Action>(item.getActions());
```

Comment:
`this.actions` is public, so I guess there could be extensions expecting transient actions to be present there. Can't find any code using this directly though.